tenderly-suite/role-update-action/role-update/roleUpdate.js

```diff
@@ -0,0 +1,102 @@
+    const chainName = context.metadata.getNetwork();
+
+    const roleAddedAccounts = [];
+    const addedRoles = [];
+    const roleRemovedAccounts = [];
+    const removedRoles = [];
+    const summary = 'Roles updated';
+    let severity = 0;
+
+    for (const log of event.logs) {
+        if (log.topics[0] === TOPIC_0_ROLES_ADDED || log.topics[0] === TOPIC_0_ROLES_REMOVED) {
+            console.log('log Found');
+            const length = parseInt(log.data.substring(128, 130), 16);
+            const roles = [];
+
+            for (let index = 0; index < length; index++) {
+                const subIndex = 64 * (3 + index) + 2;
+                roles.push(getRole(parseInt(log.data.substring(subIndex - 2, subIndex), 16)));
+            }
+
+            const account = `0x${log.topics[1].substring(26, 26 + 40)}`;
+            const tempSeverity = TRUSTED_ADDRESSES.includes(account.toLowerCase()) ? 1 : 2;
+
+            if (log.topics[0] === TOPIC_0_ROLES_ADDED) {
+                roleAddedAccounts.push(account);
+                addedRoles.push(roles);
+            } else {
+                roleRemovedAccounts.push(account);
+                removedRoles.push(roles);
+            }
+
+            if (tempSeverity > severity) {
+                severity = tempSeverity;
+            }
+        }
+    }
+
+    if (severity) {
+        try {
+            await axios.post(
+                PAGER_DUTY_ALERT_URL,
+                {
+                    routing_key: await context.secrets.get('PD_ROUTING_KEY'),
+                    event_action: 'trigger',
+                    payload: {
+                        summary,
+                        source: `${chainName}-${event.hash}`,
+                        severity: severity === 2 ? 'warning' : 'info',
+                        custom_details: {
+                            timestamp: Date.now(),
+                            chain_name: chainName,
+                            trigger_event: event,
+                            rolesUpdated: {
+                                roleAddedAccounts,
+                                addedRoles,
+                                roleRemovedAccounts,
+                                removedRoles,
+                            },
+                        },
+                    },
+                },
+                {
+                    'Content-Type': 'application/json',
+                },
+            );
+        } catch (error) {
+            console.log('PD error status: ', error.response.status);
+            console.log('PD error data: ', error.response.data);
+            throw Error('SENDING_ALERTS_FAILED');
+        }
+    } else {
+        throw new Error('NO_ROLE_UPDATES_DETECTED');
+    }
+};
+
+function getRole(roleId) {
+    if (roleId === 0) {
+        return 'Distributor';
+    } else if (roleId === 1) {
+        return 'Operator';
+    } else if (roleId === 2) {
+        return 'FlowLimiter';
+    }
+
+    return '-';
```

Comment:
Maybe add some sort of error or warning explaining that the `roleId` is not recognized, just returning '-' does not give me much context unless this is a specific design decision you made